client/core/trade.go

```diff
@@ -354,6 +360,14 @@ func (t *trackedTrade) negotiate(msgMatches []*msgjson.Match) error {
 		if oid != t.ID() {
 			return fmt.Errorf("negotiate called for wrong order. %s != %s", oid, t.ID())
 		}
+		// Do not process matches with existing matchTrackers. e.g. In case we
+		// start "extra" matches from the 'connect' response negotiating via
+		// authDEX>readConnectMatches, and a subsequent resent 'match' request
+		// leads us here again or vice versa. Or just duplicate match requests.
+		if t.matches[mid] != nil {
+			log.Warnf("Skipping match %v that is already negotiating.", mid)
+			continue
+		}
```

Comment:
I don't see a way this could happen at present, but starting up a new matchTracker and overwriting an existing one would be bad news.